src/shogun/features/DenseFeatures.cpp

```diff
@@ -977,101 +1007,66 @@ template<class ST> bool CDenseFeatures<ST>::is_equal(CDenseFeatures* rhs)
 	return true;
 }
 
-template<class ST> CFeatures* CDenseFeatures<ST>::create_merged_copy(
-		CList* others)
+template <class ST>
+CFeatures* CDenseFeatures<ST>::create_merged_copy(CList* others)
 {
-	SG_DEBUG("entering %s::create_merged_copy()\n", get_name());
+	SG_DEBUG("Entering.\n");
 
-	if (!others)
-		return NULL;
+	REQUIRE(others!=nullptr, "The list of feature instances cannot be null!\n");
 
-	/* first, check other features and count number of elements */
-	CSGObject* other=others->get_first_element();
-	index_t num_vectors_merged=num_vectors;
-	while (other)
+	auto current=others->get_first_element();
+	auto total_num_vectors=get_num_vectors();
+	auto unref_required=others->get_delete_data();
+
+	while (current!=nullptr)
 	{
-		CDenseFeatures<ST>* casted=dynamic_cast<CDenseFeatures<ST>* >(other);
+		auto casted=dynamic_cast<CDenseFeatures<ST>*>(current);
 
-		if (!casted)
-		{
-			SG_ERROR("%s::create_merged_copy(): Could not cast object of %s to "
-					"same type as %s\n",get_name(), other->get_name(), get_name());
-		}
+		REQUIRE(casted!=nullptr, "Could not cast object of %s to same type as %s!\n",
```

Comment:
Same here, make it user understandable.
"Provided object's type (%s) at index (%d) in list must match own type (%s)